src/git_changelog/build.py

```diff
@@ -338,79 +360,98 @@ def parse_commits(self) -> list[Commit]:
-                        next_version.compare_url = self.provider.get_compare_url(
-                            base=version.tag,
-                            target=next_version.tag or "HEAD",
-                        )
-                next_version = version
-                versions_list.append(version)
-                versions_types_dict[commit.version] = {}
-            versions_dict[commit.version].commits.append(commit)
-            if "type" in commit.convention and commit.convention["type"] not in versions_types_dict[commit.version]:
-                section = Section(section_type=commit.convention["type"])
-                versions_types_dict[commit.version][commit.convention["type"]] = section
-                versions_dict[commit.version].sections_list.append(section)
-                versions_dict[commit.version].sections_dict = versions_types_dict[commit.version]
-            versions_types_dict[commit.version][commit.convention["type"]].commits.append(commit)
-        if next_version is not None and self.provider:
-            next_version.compare_url = self.provider.get_compare_url(
-                base=versions_list[-1].commits[-1].hash,
-                target=next_version.tag or "HEAD",
-            )
         return versions_list, versions_dict
 
+    def _create_version(self, commit: Commit) -> Version:
+        date = commit.committer_date.date() if commit.version else datetime.date.today()  # noqa: DTZ011
+        version = Version(tag=commit.version, date=date)
+        if self.provider:
+            version.url = self.provider.get_tag_url(tag=commit.version)
+        return version
+
+    def _assign_previous_versions(self) -> None:
+        """Assign each version its previous version and create the compare URL.
+
+        The previous version is defined as the first version, that is found
+        by following the left branch of the commit graph.
+
+        If no previous version is found, either because it is the first commit or
+        due to the commit filter excluding it, the compare URL is created with the
+        first commit (oldest).
+        """
+        for version in self.versions_list:
+            next_commit = version.commits[0]
+            previous_version: str = ""
+            while not previous_version:
+                parent_commits = next_commit.parent_commits
+                if not parent_commits:
+                    previous_version = next_commit.hash
+                else:
+                    # Only follow the left branch of the commit tree.
+                    next_commit = parent_commits[0]
+                    if next_commit.tag:
+                        previous_version = next_commit.tag
+            if self.provider:
+                version.compare_url = self.provider.get_compare_url(
+                    base=previous_version,
+                    target=version.tag or "HEAD",
+                )
+                version.previous_version = self.versions_dict.get(previous_version)
```

Comment:
I think assigning the previous version does not depend the presence of a provider right?

```suggestion
            version.previous_version = self.versions_dict.get(previous_version)
```